api/v1beta1/foundationdbcluster_types.go

```diff
@@ -640,6 +640,30 @@ const (
 	MissingProcesses ProcessGroupConditionType = "MissingProcesses"
 )
 
+// GetProcessGroupConditionType returns the ProcessGroupConditionType for the matching string or an error
+func GetProcessGroupConditionType(processGroupConditionType string) (ProcessGroupConditionType, error) {
```

Comment:
Could we do this through reflection instead of a switch? That might make it easier to add more conditions in the future.

Reply:
Good point, I take look 👍

Reply:
I would leave it as it is and we could refactor this in a later step. We probably have to revisit this anyway once we move to the FoundationDBProcessGroup.